modules/order/src/Adjustment.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\commerce_order;
+use Drupal\commerce_price\Price;
+
+/**
+ * Represents an adjustment.
+ */
+final class Adjustment {
+
+  /**
+   * The adjustment type.
+   *
+   * @var string
+   */
+  protected $type;
+
+  /**
+   * The adjustment label.
+   *
+   * @var string
+   */
+  protected $label;
+
+  /**
+   * The decimal amount.
+   *
+   * @var string
+   */
+  protected $amount;
+
+  /**
+   * The currency code.
+   *
+   * @var string
+   */
+  protected $currencyCode;
+
+  /**
+   * The source identifier of the adjustment.
+   *
+   * @var string
+   */
+  protected $sourceId;
+
+  /**
+   * Constructs a new Adjustment instance.
+   *
+   * @param array $definition
+   *   The definition.
+   */
+  public function __construct(array $definition) {
+    foreach (['type', 'label', 'amount', 'currency_code', 'source_id'] as $required_property) {
+      if (empty($definition[$required_property])) {
+        throw new \InvalidArgumentException(sprintf('Missing required property %s.', $required_property));
+      }
+    }
+
+    $this->type = $definition['type'];
```

Comment:
This could be a pattern we're using elsewhere, but I'm curious why we aren't setting these values above in the foreach. e.g. $this->{$required_property} = $definition[$required_property];

Perhaps because we also want to initialize optional default properties in other classes?